### PR TITLE
create new method to retrieve Entity or create a new Instance

### PR DESCRIPTION
**Pull request by nedimdzendzo**

### CODE REVIEW

**Comment by trq** [comment]
:+1:

**Comment by jordonsc** [comment]
:+1:

File: src/Bravo3/Orm/Services/EntityManager.php [hunk 1]
```diff
@@ -330,6 +330,23 @@ public function retrieveEntityOrNull($class_name, $id, $use_cache = true)
...
+     * @param int    $id
+     * @param bool   $use_cache
+     * @return object
+     */
+    public static function retrieveEntityOrNew($class_name, $id, $use_cache = true)
+    {
+        try {
+            return $this->retrieve($class_name, $id, $use_cache);
+        } catch (NotFoundException $e) {
+            return new $class_name;
```
Comment (purinda) [inline src/Bravo3/Orm/Services/EntityManager.php:345]: @nedimdzendzo no need to pass the $id in as a parameter?

**Comment by purinda** [comment]
:+1:

File: src/Bravo3/Orm/Services/EntityManager.php (hunk 1 above)
Comment (jordonsc) [inline src/Bravo3/Orm/Services/EntityManager.php:345]: Curious how you plan to retrieve an existing entity then?

**Pull request merged**